examples/solitaire/src/SolitairePuzzle.tsx

```diff
@@ -106,29 +109,41 @@ export const Puzzle = ({
         cards.push(column);
         currentIndex += i;
       }
+
+      const newState: HistoryState = {
+        flippedCards: [],
+        stockCards: shuffledCards.slice(28), 
+        suitCards: state.suitCards,
+        tableauCards: cards, 
+        wasteCards: state.wasteCards,
+      };
+      setGameHistory((prev) => {
+        const state = [...(prev ?? []), newState];
+        onProgress?.(JSON.stringify(state));
+        return state;
+      });
+
       return {
         ...prev,
         stockCards: shuffledCards.slice(28),
         tableauCards: cards,
       };
-    });
+    }); 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
   
   React.useEffect(() => {
-    if (data && !startFresh && !loaded) {
+    if (loaded) {
+      return;
+    }
+    setLoaded(true);
+    if (data && !startFresh) {
       try {
-        const { state, history } = (typeof data === 'string' ? JSON.parse(data) : data) as { state: HistoryState; history: HistoryState[]; };
-        if (history.length > 0) {
-          setState(state);
-          setGameHistory(history);
-        } else {
-          throw new Error('Error loading history');
-        }
+        const history = typeof data === 'string' ? JSON.parse(data) : data;
+        setGameHistory(history);
```

Comment:
you need also set the current `state` in addition to the game history

Reply:
it also makes things weird